drivers/soundwire/mipi_disco.c

```diff
@@ -280,6 +280,35 @@ static int sdw_slave_read_dpn(struct sdw_slave *slave,
 	return 0;
 }
 
+/*
+ * In MIPI DisCo spec for SoundWire, lane mapping for a slave device is done with
+ * mipi-sdw-lane-x-mapping properties, where x is 1..7, and the values for those
+ * properties are mipi-sdw-manager-lane-x or mipi-sdw-peripheral-link-y, where x
+ * is an integer between 1 to 7 if the lane is connected to a master lane, y is a
+ * character between A to E if the lane is connected to another slave lane.
+ */
+static int sdw_slave_read_lane_mapping(struct sdw_slave *slave)
+{
+	struct sdw_slave_prop *prop = &slave->prop;
+	struct device *dev = &slave->dev;
+	char prop_name[30];
+	const char *prop_val;
+	size_t len;
+	int ret, i;
+
+	for (i = 0; i < SDW_MAX_LANES; i++) {
+		snprintf(prop_name, sizeof(prop_name), "mipi-sdw-lane-%d-mapping", i);
+		ret = device_property_read_string(dev, prop_name, &prop_val);
```

Comment:
why read a string if you are trying to read an hex value?

Reply:
That's the dilemma here, the possible values for the property `mipi-sdw-lane-x-mapping` are `mipi-sdw-manager-lane-y` or `mipi-sdw-peripheral-link-z` in the DisCo spec, where y is an integet between 1 to 7, z is an char between A to Z.

With the last character y or z, the link is able to be identified, so I just use the last character from the property value string. Here I store them as character, `'1', '2', ... '7', 'A',...,'E'`. A second step to transform them to hex value is possible.
Or we could store the intact string `mipi-sdw-manager-lane-y/mipi-sdw-peripheral-link-z` with a string array, but I think it will be hard to use later, because I think the 1..7 number also denote the lane ID for master device, parsing is needed later.

Reply:
> why read a string if you are trying to read an hex value?

Because the value shall be `mipi-sdw-manager-lane-<m>`, where <m> is an integer between 1 and 7 if the lane connects to Manager, or `mipi-sdw-peripheral-link-<tag>`, where <tag> is 1 to 5 characters that uniquely identify the Link in the system if the line connects to Peripheral.

Reply:
> I would report a warning on mipi-sdw-peripheral-link- and ignore it because we don't have support for device-to-device communication. 

This is good to me, currently, I am asking Mengdong's help to find PTL BIOS owner, to align the property definition in BIOS. I will update the PR after the alignment.